src/lib/sanitize.js

```diff
@@ -0,0 +1,103 @@
+/*****
+ License
+ --------------
+ Copyright © 2020-2025 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Mojaloop Foundation for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Mojaloop Foundation
+ * Kevin Leyow <kevin.leyow@infitx.com>
+
+ --------------
+ ******/
+
+/**
+ * Sanitizes a request object for safe logging by redacting sensitive fields.
+ * Fields such as Authorization, httpAgent, and httpsAgent are replaced with '[REDACTED]'.
+ * Note: This function mutates the original request object.
+ * @param {Object} request - The request object to sanitize.
+ * @returns {Object} The sanitized request object.
+ */
+function sanitizeRequest(request) {
+    if (!request || typeof request !== 'object') return request;
+
+    if (request.headers && request.headers.Authorization) {
+        request.headers.Authorization = '[REDACTED]';
+    }
+    if (request.httpAgent) request.httpAgent = '[REDACTED]';
+    if (request.httpsAgent) request.httpsAgent = '[REDACTED]';
+    return request;
```

Comment:
Better not mutate the request. Instead we can:
```js
return {
  ...request,
  ...request.headers?.Authorization && {headers: {...request.headers, Authorization: '[REDACTED]'}},
  ...request.httpAgent && {httpAgent: '[REDACTED]',
  ...request.httpsAgent && {httpsAgent: '[REDACTED]'
}
```
Please test. I am a bit surprised the headers are not lowercase.